src/object/coin_explode.cpp

```diff
@@ -18,19 +18,52 @@
 
 #include "math/random.hpp"
 #include "object/coin.hpp"
+#include "sprite/sprite.hpp"
+#include "sprite/sprite_manager.hpp"
 #include "supertux/sector.hpp"
 
 #include <list>
 
 CoinExplode::CoinExplode(const Vector& pos, bool count_stats, const std::string& sprite) :
+  GameObject("coin-explode"),
   m_sprite(sprite),
   position(pos),
   m_count_stats(count_stats)
 {
 }
 
+CoinExplode::CoinExplode(const ReaderMapping& reader) :
+  GameObject(reader),
+  m_sprite("images/objects/coin/coin.sprite"),
+  position(),
+  m_count_stats(true)
+{
+  reader.get("x", position.x);
+  reader.get("y", position.y);
+  bool emerge = false;
+}
+
+void
+CoinExplode::update(float dt_sec)
+{
+  if (is_grabbed()) 
+    // Don't do anything while being carried
+    return;
```

Comment:
This statement looks equivalent to an empty method? Is this part work in progress?